src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/IncrementalBuildTest.cs

```diff
@@ -514,6 +514,71 @@ public void ProduceReferenceAssembly ()
 			}
 		}
 
+		[Test]
+		public void TransitiveDependencyProduceReferenceAssembly ()
+		{
+			var path = Path.Combine (Root, "temp", TestName);
+			var app = new XamarinAndroidApplicationProject {
+				ProjectName = "App",
+				//NOTE: so _BuildApkEmbed runs in commercial tests
+				EmbedAssembliesIntoApk = true,
+				AndroidUseSharedRuntime = false,
```

Comment:
Does this test still work if you remove these settings? Just so it will have fast deployment enabled like most apps by default.